### ISSUE TITLE
Create: Safari Extension

### ISSUE THREAD

**Issue by scysys** [marius-wieschollek/passwords-webextension#26]

**Comment by marius-wieschollek** [comment]
Can you please add more details and context to this issue?

**Comment by scysys** [comment]
You already created Browser Extensions for "Firefox / Chromium". So create another Extension for Safari

**Comment by marius-wieschollek** [comment]
TL;DR: I will not make a Safari Extension because Apple has decided to make this a long and expensive misadventure.

Since extensions are a common feature of modern browsers, all major browsers support a [similar api](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions) in order to make the development of cross browser extensions easier. Apple on the other side decided to integrate its own poorly documented api for safari which requires me to put in a lot of additional work and create an extension just for Safari if i wanted to support it.

Safari, just like Microsoft Edge, runs only on one operating system: macOs. Microsoft with its 90% market share decided, that they should also provide a way to develop for Edge without having Windows and offers [free virtual machines](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/) for developers. Apple on the other hand decided that you have to buy a ridiculously overpriced macOs device to do so. Apple also has also ensured that no VM software officially supports running macOs if the host system is not an Apple device. (Even if this is technically possible.) So i would have to spend around 1,5K € to buy an Apple Laptop equipped with 200€-class hardware.

To publish an extension these days, there is basically no other option than to use the official extension repository of that browser. While Mozilla and Google (haven't checked Microsoft) provide a properly maintained store, with fast reviews and proper communication either for free or a acceptable small one time fee, Apple has something different. Apple has decided that you need to pay a 100$ yearly developer fee to publish free extensions on their [safari extension listing](https://safari-extensions.apple.com/). Not a web store with proper presentation, a listing! And what do you get for those 100$? Apparently the most intransparent and slowest possible review process if you believe the [many stories](https://medium.com/@honestbleeps/what-apple-gives-you-for-100-as-a-safari-extension-developer-and-why-reddit-enhancement-suite-6e2d829c2e52) on the internet about this.

On top of that come the many quirks of Safari that i would have to work around and the slow implementation of new standards by Apple.

Given that passwords is a rather small app and the amount of Safari users will most probably be a one digit number, i do not see any way to create an extension for Safari with a justifiable investment of time and money for me.

_In contrast to Apple, I actually provide a [developer documentation](https://git.mdns.eu/nextcloud/passwords/wikis/Developers/Index) for my app. So if anyone wishes to build a Safari Extension for Passwords, i will do what i can to support him/her._
